Composer/packages/client/src/recoilModel/dispatchers/qna.ts

```diff
@@ -89,22 +90,31 @@ export const qnaDispatcher = () => {
 
   const importQnAFromUrls = useRecoilCallback(
     (callbackHelpers: CallbackInterface) => async ({ id, urls }: { id: string; urls: string[] }) => {
-      const { set, snapshot } = callbackHelpers;
+      const { snapshot } = callbackHelpers;
       const qnaFiles = await snapshot.getPromise(qnaFilesState);
+      const projectId = await snapshot.getPromise(projectIdState);
       const qnaFile = qnaFiles.find((f) => f.id === id);
-      set(qnaAllUpViewStatusState, QnAAllUpViewStatus.Loading);
+
+      const notificationId = await addNotificationInternal(callbackHelpers, getQnAPending(urls));
```

Comment:
With my proposed change, this could look like:

```ts
const notification = createNotification(getQnAPending(urls));
await addNotificationInternal(callbackHelpers, notification);

// later
deleteNotificationInternal(callbackHelpers, notification.id);
```